### DISCUSSION TITLE
ServoEase and ESP32

### DISCUSSION THREAD

**Discussion by lamello** [ArminJo/ServoEasing#35] [General]
Hellol Armin,

Thanks for writing and sharing this code. 
I am writing code for controlling leds and servo's with an ESP32 with the PCA9685.
I can't use our lib 1 on 1, so i copied and renamed it and modified it to sue my purpose.
I removed the 20ms Ticker and do everything with de round robin update. 
I read in the code you had some issues with i2c and PCA9685. Well, i had some too.
After examine the i2c signals with an oscolloscope i saw the signals were bad. I found out that the wire lib of the ESP32 use the internal pullup resisters as the i2c pullup's. They are to high to be in spec of the 12c specs. I have removed the PULLUP option in the wire library and use 2k2 resistors as i2c pullups.
The i2c signals are now good and i use the 400000hrz i2c busfrequency. I works now for 8 hours without problems.

Again, thanks for the lib.

Greets lamello

**Comment by ArminJo** [comment]
Thanks for the testing.
I am documenting this in the source now and am wondering if it also works without removing the pullup option and if it works with 800 kHz?
Do you have done some tests regarding this?

**Reply by lamello** [reply]
Hello Arwin,

I have re-enable the pullup's in the core i2c code and use the 800kHz i2c busfrequency. I still use the 2k external pullup resistors. 
It is now testing for over a hour without a problem. 
Even my roundtrip time has reduced from 2 to 1 ms.
It great's that you are working on documentation. I would appreciate some information about those different moving modes and the values that are in rang for which modes.

I have measured the value of the internal resistor, it's about 25k. That's much to weak for an i2c pullup.

Greets Lamello

**Reply by ArminJo** [reply]
I am very sorry, but I tested it and it does not work, if you use the timer like in the example. Only 100 kHz works 😢 .

**Reply by lamello** [reply]
Hello Arwin,

That's why i killed the Timer Tick. Without the timer and do the update in the loop(), it works like a charm.

Greets Lamello
